dataprofiler/tests/profilers/test_profile_builder.py

```diff
@@ -1444,5 +1444,82 @@ def test_null_row_stats_correct_after_updates(self, *mocks):
         self.assertSetEqual({8}, profile._profile[1].null_types_index['None'])
 
 
+class TestProfilerFactoryClass(unittest.TestCase):
+
+    def test_profiler_factory_class_bad_input(self):
+        with self.assertRaisesRegex(ValueError, "Must specify 'profiler_type' "
+                                                "to be 'structured' or "
+                                                "'unstructured'."):
+            Profiler(pd.DataFrame([]), profiler_type="whoops")
+
+        with self.assertRaisesRegex(ValueError, "Data must either be imported "
+                                                "using the data_readers, "
+                                                "pd.Series, or pd.DataFrame."):
+            Profiler("whoops")
+
+    @mock.patch('dataprofiler.profilers.profile_builder.'
+                'ColumnPrimitiveTypeProfileCompiler')
+    @mock.patch('dataprofiler.profilers.profile_builder.'
+                'ColumnStatsProfileCompiler')
+    @mock.patch('dataprofiler.profilers.profile_builder.'
+                'ColumnDataLabelerCompiler')
+    @mock.patch('dataprofiler.profilers.profile_builder.DataLabeler')
+    @mock.patch('dataprofiler.profilers.profile_builder.UnstructuredCompiler')
+    def test_profiler_factory_class_creates_correct_profiler(self, *mocks):
+        """
+        Ensure Profiler factory class either respects user input or makes
+        reasonable inference in the absence of user specificity.
+        """
+        # User specifies via profiler_type
+        empty_df = pd.DataFrame([])
+        empty_ser = pd.Series([])
+        self.assertIsInstance(Profiler(empty_df, profiler_type="structured"),
+                              StructuredProfiler)
+        self.assertIsInstance(Profiler(empty_df, profiler_type="unstructured"),
+                              UnstructuredProfiler)
+
+        # User gives data that has .is_structured == True
+        empty_csv_df = dp.Data(data=empty_df, data_type="csv")
+        self.assertIsInstance(Profiler(empty_csv_df), StructuredProfiler)
+
+        # User gives data that has .is_structured == False
+        empty_csv_rec = dp.Data(data=empty_df, data_type="csv",
+                                options={"data_format": "records"})
+        self.assertIsInstance(Profiler(empty_csv_rec), UnstructuredProfiler)
```

Comment:
this is fine the way it is, but if we wanted to not be dependent on readers, could do:
```python
empty_struct_data = mock.Mock(spec=dp.data_readers.base_data.BaseData)
empty_struct_data.is_structured = True
```

and likewise for unstructured